src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,40 @@
 package core.mate.academy.service;
 
-/**
- * Your implementation of MachineService.
- */
-public class MachineServiceImpl {
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.BulldozerProducer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.ExcavatorProducer;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import core.mate.academy.model.TruckProducer;
+import java.util.ArrayList;
+import java.util.List;
+
+public class MachineServiceImpl implements MachineService<Machine> {
+
+    @Override
+    public List getAll(Class<? extends Machine> type) {
+        if (type.equals(Bulldozer.class)) {
+            return new BulldozerProducer().getListMachines();
+        } else if (type.equals(Excavator.class)) {
+            return new ExcavatorProducer().getListMachines();
+        } else if (type.equals(Truck.class)) {
+            return new TruckProducer().getListMachines();
+        }
+        return new ArrayList<>();
+    }
```

Comment:
maybe will be better if the last block will be else and not necessarily return empty List